pkg/services/authn/clients/api_key.go

```diff
@@ -173,6 +174,27 @@ func (s *APIKey) Hook(ctx context.Context, identity *authn.Identity, r *authn.Re
 	return nil
 }
 
+func (s *APIKey) getAPIKeyID(ctx context.Context, identity *authn.Identity, r *authn.Request) (apiKeyID int64, exists bool) {
```

Comment:
I have opted in for using `bool` instead of error, as the call is being done in the hook, which we can safely ignore. Additionally, at this point the request has been already validated, so if anything the API key should be valid to fetch.

So I thought just logging as a warning should suffice.